integration-cli/docker_cli_update_unix_test.go

```diff
@@ -187,3 +176,25 @@ func (s *DockerSuite) TestUpdateStats(c *check.C) {
 	c.Assert(preMemLimit, checker.Equals, curMemLimit)
 
 }
+
+func (s *DockerSuite) TestUpdateRestartPolicy(c *check.C) {
+	out, _ := dockerCmd(c, "run", "-d", "--restart=on-failure:3", "busybox", "sh", "-c", "sleep 1 && false")
+	timeout := 15 * time.Second
+	if daemonPlatform == "windows" {
+		timeout = 50 * time.Second
+	}
+
+	id := strings.TrimSpace(string(out))
+
+	// update restart policy to on-failure:5
+	dockerCmd(c, "update", "--restart=on-failure:5", id)
+
+	err := waitInspect(id, "{{ .State.Restarting }} {{ .State.Running }}", "false false", timeout)
```

Comment:
Hmm, checking `RestartCount` hits 5 may bring new problem, if `docker update` fails to update the `RestartPolicy` of a running container, it will never hits 5...Besides that, we still need to `waitInspect` the `RestartCount` in timeout.

I'll try to give it a larger timeout, maybe 60 seconds? If the container can't stop in 60 seconds, I think some real problem happens.